src/server/pfs/server/driver.go

```diff
@@ -1085,84 +1081,103 @@ func (d *driver) propagateBranches(ctx context.Context, txnCtx *txncontext.Trans
 	return nil
 }
 
+func (d *driver) inspectCommitWithID(ctx context.Context, commit *pfs.Commit, wait pfs.CommitState) (*pfsdb.CommitWithID, error) {
```

Comment:
This is not great that we are duplicating the inspect commit functionality for the `pfs.Commit` / `pfsdb.CommitWithID`. The core functionality should be in one function.

Reply:
That's an oversight on my end. I can structure this to call inspectCommit()